pkg/build/builder/common.go

```diff
@@ -66,6 +66,25 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildStatus(c client.BuildInterface, build *api.Build) {
+	// NOTE: if reason is empty there's nothing to update, do i need this check
+	// here?
+	if build.Status.Reason == "" {
+		return
+	}
+
+	switch build.Status.Reason {
+	case api.StatusReasonPostCommitHookFailed:
+		build.Status.Message = "failed because of post commit hook"
```

Comment:
@mfojtik because i'd rather see the Reason value and the Message value defined in the same place for easy updating, and someone's going to have to write a test that looks for these values, so they're going to want constants to reference.